core/src/apps/cardano/seed.py

```diff
@@ -13,20 +13,58 @@
 
 
 class Keychain:
-    """Cardano keychain hard-coded to SEED_NAMESPACE."""
+    """Cardano keychain hard-coded to SEED_NAMESPACE_BYRON and SEED_NAMESPACE_SHELLEY."""
 
     def __init__(self, root: bip32.HDNode) -> None:
         self.root = root
+        self.byron_root = None
+        self.shelley_root = None
+
+        self.byron_namespace_length = len(SEED_NAMESPACE_BYRON)
```

Comment:
why having those constants as class properties?

Reply:
True, it was a bit weird. Fixed.